### PR TITLE
Fix future/pageable types.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
The previous LRO changes were incomplete, they missed the case where
futures intersect with pageable types.
Avoid importing the net/http package when it's not needed.

### CODE REVIEW

File: src/Model/CodeModelGo.cs [hunk 1]
```diff
@@ -137,10 +137,19 @@ public IEnumerable<string> ModelImports
                 {
                     imports.Add(PrimaryTypeGo.GetImportLine("github.com/Azure/go-autorest/autorest"));
                 }
-                if (ModelTypes.Any(mt => mt is FutureTypeGo))
+                var futureModels = ModelTypes.Where(mt => mt is FutureTypeGo).Cast<FutureTypeGo>();
+                if (futureModels.Any())
```
Comment (vladbarosan) [inline src/Model/CodeModelGo.cs]: not sure what was wrong with the previous if, also why do we cast to FutureTypeGo

Comment (vladbarosan) [reply src/Model/CodeModelGo.cs]: oh ok nevermind we need the collection later, but do we need the cast ?

---
In reply to: [192249868](https://github.com/Azure/autorest.go/pull/130#discussion_r192249868) [](ancestors = 192249868)

Comment (jhendrixMSFT) [reply src/Model/CodeModelGo.cs]: Yes, the ```IsDefaultReturnType``` property is only on ```FutureTypeGo```.

File: src/Model/CodeModelGo.cs [hunk 2]
```diff
@@ -137,10 +137,19 @@ public IEnumerable<string> ModelImports
...
                 }
-                if (ModelTypes.Any(mt => mt is FutureTypeGo))
+                var futureModels = ModelTypes.Where(mt => mt is FutureTypeGo).Cast<FutureTypeGo>();
+                if (futureModels.Any())
                 {
                     imports.Add(PrimaryTypeGo.GetImportLine("github.com/Azure/go-autorest/autorest/azure"));
-                    imports.Add(PrimaryTypeGo.GetImportLine("net/http"));
+                    // if any of the futures return the non-default response then we need the net/http package
+                    foreach (var ftg in futureModels)
+                    {
```
Comment (vladbarosan) [inline src/Model/CodeModelGo.cs]: maybe ```if(futureModels.Any( fm => !fm.IsDefaultReturnType)) {}```

**Pull request merged**